backend/models/volunteer_model.py

```diff
@@ -0,0 +1,48 @@
+import calendar
+import json
+from datetime import datetime as dt
+from mongoengine import (Document, StringField, IntField, EmailField, BooleanField, ListField,
+                         DateTimeField, DictField, FloatField, URLField, ReferenceField)
+from config import PassHash
+
+FACEBOOK_URL_REGEX = r"http(?:s):\/\/(?:www\.)facebook\.com\/.+"
+
+
+class User(Document):
+    # TODO: Each volunteer must have a reference to operator that was created by
+    # created_by = ReferenceField("operator")
+    first_name = StringField(max_length=50, default='No_First')
+    last_name = StringField(max_length=50, default='No_Last')
+    email = EmailField(required=True)
+    password = StringField(required=True, min_length=200)
+    phone = IntField(min_value=10000000, max_value=99999999)
+    created_at = DateTimeField(default=dt.now)
+    # we don't delete users just deactivating them
+    logins = ListField(default=[])
+
+    is_active = BooleanField(default=True)
+
+    meta = {'allow_inheritance': True}
+
+
+    def clean_data(self) -> dict:
+        data = self.to_mongo()
+        if "password" in data and data["password"]:
+            del data["password"]
+        if "logins" in data:
+            del data["logins"]
+        return data
+
+    def check_password(self, password) -> bool:
+        data = self.to_mongo()
+        return PassHash.verify(password, data["password"])
+
+
+class Volunteer(User):
```

Comment:
So far the intention is to use a Telegram bot to interact with volunteers - so we'd need a `telegram_id` field as well.